### PR TITLE
Domains: Add explanation when trying to transfer domain that is in 60 days transfer lock

### PR DESCRIPTION
**Pull request by leonardost**
### Proposed Changes

Note: This depends on D87294-code.

This PR adds some checks and explanation messages when a customer tries to transfer in a domain that's less than 60 days old, i.e. it has a transfer lock. This should help reduce support interactions caused by domain transfers that can't be done because of transfer locked domains.

See pcYYhz-OU-p2 for more context.

#### Screenshots

- Transfer or connect step
 
Markup on 2022-09-09 at 20:03:06

- Searching for a newly registered domain

Markup on 2022-09-09 at 20:18:31

- Transfer or connect step during new site creation

Markup on 2022-09-09 at 20:20:47

### Testing Instructions

- Build this branch locally or open the live Calypso link
- Apply D87294-code to your sandbox and sandbox `public-api`
- Go to "Upgrades > Domains"
- Click on "+ Add a domain" and "Use a domain I own"
- Input a domain that was recently registered in another provider
- Ensure the "transfer or connect" step has the transfer option disabled, like in the screenshots above

Please test the domain search form too:

- Go to "Upgrades > Domains" again
- Click on "+ Add a domain" and "Search for a domain"
- Search for a domain that was recently registered in another provider
- Ensure the "&lt;domain&gt; is already registered. Do you own it?" notice appears below the search box

Also please test the domain selection during site creation too:

- Click the "Add new site" button in the left nav bar in Calypso or go to `/start/domains`
- Search for a domain that was recently registered in another provider
- Ensure the "&lt;domain&gt; is already registered. Do you own it?" notice appears below the search box
- If you click the "Do you own it?" link, ensure the "transfer or connect" step has the transfer option disabled, like in the screenshots above

### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -1439,9 +1439,11 @@ class RegisterDomainStep extends Component {
 	};
 
 	showAvailabilityErrorMessage( domain, error, errorData ) {
-		const { DOTBLOG_SUBDOMAIN, TRANSFERRABLE } = domainAvailability;
+		const { DOTBLOG_SUBDOMAIN, TRANSFERRABLE, RECENT_REGISTRATION_LOCK_NOT_TRANSFERRABLE } =
+			domainAvailability;
 		if (
 			( TRANSFERRABLE === error && this.state.lastDomainIsTransferrable ) ||
+			RECENT_REGISTRATION_LOCK_NOT_TRANSFERRABLE === error ||
```
Comment (leonardost) [inline client/components/domains/register-domain-step/index.jsx:1450]: This additional condition is to not show an error notice if a recently registered domain is searched. In these cases, we should only show the "<domain> is already registered. Do you own it?" notice.

**Review by rafaelgallani — APPROVED** [review]
image

LGTM! Nice job handling this one @leonardost! 
I can confirm that all test cases are working as expected.

**Comment by rafaelgallani** [comment]
There's only one thing I noticed - totally unrelated to these changes... But just so we're aware of it:

When clicking the "Do you own it?" link in the `/start/domains` flow, the "Use a domain I own" component gets loaded with the correct domain, but it stays in its initial state - it doesn't load the Connect/Transfer options. Here's a recording of it:

https://user-images.githubusercontent.com/18705930/189798758-6020eda2-7d80-4065-8190-6d04fc5e98c8.mov

It also happens on `trunk`. Maybe, in the future, we could write some tests for these new components... It would surely save us some time.. 😅

**Comment by leonardost** [comment]
Thanks for the review @rafaelgalani!

> When clicking the "Do you own it?" link in the `/start/domains` flow, the "Use a domain I own" component gets loaded with the correct domain, but it stays in its initial state - it doesn't load the Connect/Transfer options

I noticed that too 🤔 it would be more natural if it went straight to the next step or automatically changed to it

> we could write some tests for these new components

💯% agree 🙂

**Comment by delputnam** [comment]
@leonardost  I think that this looks good, but we may want to change the wording in the notice depending on how we decide to handle the check in the back-end patch.

File: client/components/domains/use-my-domain/utilities/get-option-info.js [hunk 2]
```diff
@@ -126,6 +126,36 @@ export function getOptionInfo( {
...
+			};
+			break;
+		case domainAvailability.SERVER_TRANSFER_PROHIBITED_NOT_TRANSFERRABLE:
+			transferContent = {
+				...optionInfo.transferNotSupported,
+				topText: createInterpolateElement(
+					sprintf(
+						/* translators: %s - the domain the user wanted to transfer */
+						__(
+							"<strong>%s</strong> can't be transferred due to a domain lock, but you can connect it instead."
```
Comment (delputnam) [inline client/components/domains/use-my-domain/utilities/get-option-info.js]: To help the user understand what's going on here, I wonder if we should be more specific and say something like "...due to a transfer lock at the registry, ..."

Comment (leonardost) [reply client/components/domains/use-my-domain/utilities/get-option-info.js]: I like that 👍 it should make things a little more clear.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7542097

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7542097).

**Pull request merged**